packages/editor/src/lib/editor/managers/WatermarkManager.ts

```diff
@@ -20,13 +20,14 @@ export class WatermarkManager {
 		if (!license.isLicenseValid) {
 			return true
 		}
-		if (license.isLicenseValid) {
-			if (!license.isDomainValid) {
-				return true
-			}
-			if (license.isLicenseExpired) {
-				return true
+		if (!license.isDomainValid) {
+			return true
+		}
+		if (license.isPerpetualLicenseExpired || license.isAnnualLicenseExpired) {
+			if (license.isInternalLicense) {
+				throw new Error('Internal license expired.')
```

Comment:
i did this internal bit slightly differently (in Editor.ts) - it's not the biggest deal but it feels like it shouldn't be in the watermark logic maybe. but i also don't care too much :P

Reply:
Yeah, good call. We have a couple of PRs in flight, but I'll add an issue to our project to revisit this before it goes out.